src/Guzzle/StatusDateOverrideMiddleware.php

```diff
@@ -0,0 +1,113 @@
+
+                    case 'application/vnd.elife.article-list+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+
+                    case 'application/vnd.elife.article-related+json; version=1':
+                        $data = $this->updateItems($data);
+                        break;
+
+                    case 'application/vnd.elife.article-poa+json; version=1':
+                    case 'application/vnd.elife.article-poa+json; version=2':
+                    case 'application/vnd.elife.article-vor+json; version=1':
+                    case 'application/vnd.elife.article-vor+json; version=2':
+                    case 'application/vnd.elife.article-vor+json; version=3':
+                        $data = $this->updateItem($data);
+                        break;
+
+                    case 'application/vnd.elife.collection+json; version=1':
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.cover-list+json; version=1':
+                    case 'application/vnd.elife.highlight-list+json; version=3':
+                        $data['items'] = array_map(function (array $cover) {
+                            $cover['item'] = $this->updateItem($cover['item']);
+
+                            return $cover;
+                        }, $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.podcast-episode+json; version=1':
+                        $data['chapters'] = array_map(function (array $chapter) {
+                            $chapter['content'] = $this->updateItems($chapter['content'] ?? []);
+
+                            return $chapter;
+                        }, $data['chapters']);
+                        break;
+
+                    case 'application/vnd.elife.person-list+json; version=1':
+                        $data['items'] = array_map([$this, 'updatePerson'], $data['items']);
+                        break;
+
+                    case 'application/vnd.elife.press-package+json; version=3':
+                        $data['relatedContent'] = $this->updateItems($data['relatedContent'] ?? []);
+                        break;
+
+                    case 'application/vnd.elife.search+json; version=1':
+                        $data['items'] = $this->updateItems($data['items']);
+                        break;
+                }
+
+                return $response->withBody(stream_for(json_encode($data)));
+            });
+        };
+    }
+
+    private function updateItems(array $items) : array
+    {
+        return array_map([$this, 'updateItem'], $items);
+    }
+
+    private function updateItem(array $item) : array
+    {
+        if (isset($item['statusDate']) && isset($this->rdsArticles[$item['id']]['date']) && $this->authorizationChecker->isGranted('FEATURE_RDS')) {
```

Comment:
Not sure if this captures all of the relevant cases.

Reply:
Any chance that the `statusDate` presence picks up something that is not an article [snippet]? To understand whether we should also check a`type` field which can be extensive given the number of values.

Reply:
I'd like to avoid checking the type because it is a range of values and will need to be altered if we introduce new article types. But I believe that statusDate just applies to article VOR or POA. RDS companions will just be available for VOR articles but we need not filter on this status as the matching id in rds articles should only occur when the article is VOR.